File: app/src/main/java/ms/imagine/foodiemate/presenters/FbStorageRead.java
package ms.imagine.foodiemate.presenters;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;
import com.bumptech.glide.Glide;
import com.firebase.ui.storage.images.FirebaseImageLoader;
import java.io.InputStream;
import java.net.URL;


public class FbStorageRead extends FbStoragePresenter {

    public void downloadImage(Context context, String imgURL, ImageView imgView) {
        Glide.with(context /* context */)
                .using(new FirebaseImageLoader())
                .load(imagesRef.child(imgURL))
                .into(imgView);
    }

    public Drawable LoadImageFromWebOperations(String url) {
        try {
            InputStream ois = (InputStream) new URL(url).getContent();
            return Drawable.createFromStream(ois, "src name");
        } catch (Exception e) {
            return null;
        }
    }

}

File: app/src/main/java/ms/imagine/foodiemate/presenters/FbStorageRead.kt
package ms.imagine.foodiemate.presenters

import android.content.Context
import android.graphics.drawable.Drawable
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.firebase.ui.storage.images.FirebaseImageLoader
import com.google.firebase.storage.StorageReference
import java.io.InputStream
import java.net.URL


class FbStorageRead (): FbStoragePresenter() {

    fun downloadImage(context: Context, imgURL: String, imgView: ImageView){
        Glide.with(context /* context */)
                .using<StorageReference>(FirebaseImageLoader())
                .load(imagesRef.child(imgURL))
                .into(imgView)
    }

    fun LoadImageFromWebOperations(url: String): Drawable? {
        try {
            val ois = URL(url).content as InputStream
            return Drawable.createFromStream(ois, "src name")
        } catch (e: Exception) {
            return null
        }
    }

}